code/game/objects/machinery/floodlight.dm

```diff
@@ -136,6 +136,228 @@
+
+
+/obj/machinery/floodlightcombat/attackby(obj/item/I, mob/user, params)
+	var/list/lights = contents // has to be a static list..
+	if(!(ishuman(user)))
+		return FALSE
+	if(istype(I, /obj/item/light_bulb/tube))
+		if(on)
+			to_chat(user, "<span class='notice'>The [src]'s safety systems won't let you open the light hatch! You should turn it off first.")
+			return FALSE
+		if(lights.len > 3)
+			to_chat(user, "<span class='notice'>All the light sockets are occupied!")
+			return FALSE
+		to_chat(user, "You insert the [I] into the [src]")
+		visible_message("[user] inserts the [I] into the [src]")
+		user.drop_held_item()
+		I.forceMove(src)
+	if(istype(I, /obj/item/lightreplacer))
+		if(on)
+			to_chat(user, "<span class='notice'>The [I] cannot dispense lights into functioning machinery!")
+			return FALSE
+		if(lights.len > 3)
+			to_chat(user, "<span class='notice'>All the light sockets are occupied!")
+			return FALSE
+		var/obj/item/lightreplacer/A = I
+		if(A.CanUse())
+			A.Use(user)
+		else
+			return FALSE
+		var/obj/E = new /obj/item/light_bulb/tube
+		E.forceMove(src)
+	calculate_brightness()
+	update_icon()
+
+/// Checks each light if its working and then adds it to the total brightness amount.
+/obj/machinery/floodlightcombat/proc/calculate_brightness()
+	brightness = 0
+	for(var/obj/item/light_bulb/tube/T in contents)
+		if(T.status == 0)
+			brightness += 4
+
+/// Updates each light
+/obj/machinery/floodlightcombat/update_overlays()
+	. = ..()
+	var/offsetX
+	var/offsetY
+	/// Used to define which slot is targeted on the sprite and then adjust the overlay.
+	var/target_slot = 1
+	if(on && brightness > 0)
+		. += image('icons/obj/machines/floodlight.dmi', src, "floodlightcombat_lighting_glow", layer + 0.01, NORTH, 0, 5)
+	for(var/obj/item/light_bulb/tube/target in contents)
+		switch(target_slot)
+			if(1)
+				offsetX = 0
+				offsetY = 0
+			if(2)
+				offsetX = 6
+				offsetY = 0
+			if(3)
+				offsetX = 0
+				offsetY = 5
+			if(4)
+				offsetX = 6
+				offsetY = 5
+		. += image('icons/obj/machines/floodlight.dmi', src, "floodlightcombat_[target.status ? "brokenlight" : "workinglight"]", layer, NORTH, offsetX, offsetY)
+		target_slot++
+
+/// Called whenever someone tries to turn the floodlight on/off
+/obj/machinery/floodlightcombat/proc/switch_light()
+	if(machine_stat & (NOPOWER|BROKEN))
+		visible_message("<span class='notice'>You flip the switch , but nothing happens , perhaps its not powered?.")
+		return FALSE
+	if(!anchored || tipped)
+		visible_message("<span class='danger'>The floodlight flashes a warning led.It is not bolted to the ground.")
+		return FALSE
+	if(on)
+		on = 0
+		set_light(0, 5, "#C5E3E132")
+	else
+		on = 1
+		set_light(brightness, 5, "#C5E3E132")
+	update_icon()
+	playsound( loc, 'sound/machines/switch.ogg', 60 , FALSE)
+
+/obj/machinery/floodlightcombat/attack_hand(mob/living/user)
+	if(!ishuman(user))
+		return FALSE
+	if(user.a_intent == INTENT_GRAB)
+		var/list/lights = contents
```

Comment:
Early return

Reply:
What do you mean by early return ?  Its not meant to do all the cases at once..

Reply:
All the code after this is indented twice
Instead what you do is invert the if check and add  the switch_light() here so that you only need to use 1 indent
we do this for consistency and readable code